google-tests/soltrace_example_tests/st_powertower_sim_run_opt_test.cpp

```diff
@@ -0,0 +1,133 @@
+#include <gtest/gtest.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <sstream>
+#include <cmath>
+
+#include <string>
+#include <cstring>
+#include <vector>
+#include <exception>
+#include <cstdlib>
+
+#include "strace.h"
+#include "stapi.h"
+#include "split_csv.h"
+
+using namespace std;
+
+TEST(st_powertower_sim_run_opt_test, BasicAssertions)
+{
+	cout << "Project Directory: " << PROJECT_DIR << endl;
+	string project_dir = PROJECT_DIR;
+	
+	cout << "Current working directory: " << project_dir << endl;
+
+	string file_path = "/Power-tower-surround_singlefacet.stinput";
+	string csv_path = "/powertower_example_opt_raydata.csv";
+	
+	string input_file_path = project_dir + file_path;
+	string ground_csv_path = project_dir + csv_path;
+
+	cout << "File path: " << project_dir << endl;
+
+	std::ifstream csv_file(ground_csv_path);
+	vector<vector<string>> result = split_csv(ground_csv_path);
+
+
+	const char* file = input_file_path.data();
+	int nrays = 50000;
+	int maxrays = 5000000;
+	int seed = 1;
+	int sunshape = 0;
+	int errors = 0;
+	int powertower = 1;
+
+	int code = 0;
+
+	st_context_t cxt = ::st_create_context();
+
+	FILE* fp = fopen(file, "r");
+	if (!fp)
+	{
+		printf("failed to open system input file\n");
+		code = -1;
+	}
+
+	printf("input file: %s\n", file);
+	if (!read_system(fp, cxt))
+	{
+		printf("error in input file.\n");
+		fclose(fp);
+		code = -1;
+	}
+
+	fclose(fp);
+
+	::st_sim_params(cxt, nrays, maxrays, powertower);
+	::st_sim_errors(cxt, sunshape, errors);
+	code = ::st_sim_run(cxt, (unsigned int)seed, ::trace_progress, 0);
+
+	int Length = ::st_num_intersections(cxt);
+
+	double* x_location = new double[Length];
```

Comment:
You don't deallocate the memory after checking the values. You need to call:
```
	delete[] x_location;
```
for all of these arrays